### ISSUE TITLE
False positive for Lint/BinaryOperatorWithIdenticalOperands

### ISSUE THREAD

**Issue by obfuscoder** [rubocop/rubocop#8700]
## Expected behavior

The following expression is not a binary operation:
```
  let(:fictional_id) { rand(100) * rand(100) }
```

## Actual behavior

Rubocop reports on this with:

```
Lint/BinaryOperatorWithIdenticalOperands: Binary operator * has identical operands.
  let(:fictional_id) { rand(100) * rand(100) }
                       ^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Use code as given above.

## RuboCop version

```
0.90.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.5.5 x86_64-linux)
```

**Comment by obfuscoder** [comment]
I know I can just write rand(10000) for this. However I would still not expect this kind of report from rubocop.

**Comment by marcandre** [comment]
> The following expression is not a binary operation:

It is.

Top definition I get from google:
"A binary operator is an operator that operates on two operands and manipulates them to return a result. Operators are represented by special characters or by keywords and provide an easy way to compare numerical values or character strings."

> I know I can just write rand(10000) for this.

I agree with this 😅

**Comment by obfuscoder** [comment]
Yet the operands are not identical. Each operand is a result of a call to the same function, but those results are not equal.

**Comment by dvandersluis** [comment]
Closing this as there is no false positive, as @marcandre said. Feel free to reopen or create a new ticket if there's still an issue.

**Comment by dvandersluis** [comment]
Also this is no longer an offense as of #9221.
